costmap_2d/include/costmap_2d/layer.h

```diff
@@ -116,6 +116,18 @@ class Layer
     return false;
   }
 
+  virtual bool isStaticLayer() {
+    return false;
+  }
+
+  virtual std::shared_ptr<std::vector<double>> getDistancesFromStaticMap() {
+    return std::shared_ptr<std::vector<double>>();
```

Comment:
Good point, if the object is immutable it should probably be std::shared_ptr<const std::vector<double>>.  In either case it's a share pointer so it's not very expensive to copy.  If this is only generated once maybe it's cleaner to put in the constructor?